setup.sh

```diff
@@ -0,0 +1,239 @@
+#!/bin/bash
+
+ARCH=$(uname -m)
+LATEST_RELEASE_TAG=$(curl -s "https://api.github.com/repos/paradedb/paradedb/releases/latest" | jq -r .tag_name)
+LATEST_RELEASE_VERSION="${LATEST_RELEASE_TAG#v}"
+
+set -Eeuo pipefail
+
+installDocker() {
+  # Set default values
+  pguser="myuser"
+  pgpass="mypassword"
+  dbname="paradedb"
+
+  echo "Installing docker..."
+
+
+  echo "Which type of distro are you using?(Required to install dependecies)"
+  OPTIONS=("Debian Based" "RHEL Based" "Arch Based")
+
+
+  select opt in "${OPTIONS[@]}"
```

Comment:
We should probably support macOS here. Also, we don't need to prompt the user. We can just infer this from their system.

What we should do is:

1- When the script start, find their OS and their arch via `uname`.

2- If it's windows, exit saying it's not supported.

3- Then prompt them on which installation they want to do. If they're on macOS, say we don't have stable binaries option for macOS but they can compile from source and link to the README.

4- Install the dependencies for the relevant OS without needing to prompt them further.

5- Install ParadeDB for the relevant OS and install type without needing to prompt them further

Congratulate them and close

Reply:
We can still provide the docker image for windows right? 

I actually tried the uname approach back when I was intially writing the script but in case of linux distros we'll need to handle all possible distros that fall in these categories eg. for RHEL based we'll get Fedora, RHEL etc, similarly for Debian based distros we'll get Ubuntu, PopOS! etc..

So to avoid that I think that prompting the user would be much better.

Reply:
You're right, we should keep Windows for Docker. Got it. That makes sense to me! Thank you for explaining :)

Reply:
Cool I'll add the steps to the Readme then

Reply:
See my other review comment, I don't think any step other than a single one-liner is needed